engine/gamesys/src/gamesys/resources/res_render_prototype.cpp

```diff
@@ -53,28 +60,106 @@ namespace dmGameSystem
+                {
+                    void* res;
+                    result = dmResource::Get(factory, prototype_desc->m_RenderResources[i].m_Path, &res);
+                    if (result != dmResource::RESULT_OK)
+                        break;
+
+                    dmResource::ResourceType res_type;
+                    dmResource::GetType(factory, res, &res_type);
+                    dmRender::RenderResourceType render_resource_type = dmRender::RENDER_RESOURCE_TYPE_INVALID;
+
+                    if (res_type == res_type_material)
+                    {
+                        render_resource_type = dmRender::RENDER_RESOURCE_TYPE_MATERIAL;
+                    }
+                    else if (res_type == res_type_render_target)
+                    {
+                        render_resource_type = dmRender::RENDER_RESOURCE_TYPE_RENDER_TARGET;
+                    }
+                    else
+                    {
+                        result = dmResource::RESULT_NOT_SUPPORTED;
+                        break;
+                    }
+
+                    prototype->m_RenderResources.Push(res);
+
+                    RenderResourceMeta meta;
+                    meta.m_Name = prototype_desc->m_RenderResources[i].m_Name;
+                    meta.m_Type = render_resource_type;
+
+                    render_resource_metas.Push(meta);
+                }
             }
-            else
+
+            if (result == dmResource::RESULT_OK)
             {
-                for (uint32_t i = 0; i < prototype->m_Materials.Size(); ++i)
+                for (uint32_t i = 0; i < prototype->m_RenderResources.Size(); ++i)
                 {
-                    dmRender::AddRenderScriptInstanceMaterial(prototype->m_Instance, prototype_desc->m_Materials[i].m_Name, prototype->m_Materials[i]->m_Material);
+                    void* render_resource = prototype->m_RenderResources[i];
+                    uint64_t render_resource_val = 0;
+
+                    switch(render_resource_metas[i].m_Type)
+                    {
+                        case dmRender::RENDER_RESOURCE_TYPE_RENDER_TARGET:
+                            render_resource_val = (uint64_t) ((dmGameSystem::TextureResource*) render_resource)->m_Texture;
+                            break;
+                        case dmRender::RENDER_RESOURCE_TYPE_MATERIAL:
+                            render_resource_val = (uint64_t) render_resource;
+                            break;
+                    }
+
+                    dmRender::AddRenderScriptInstanceRenderResource(
+                        prototype->m_Instance,
+                        render_resource_metas[i].m_Name,
+                        render_resource_val,
+                        render_resource_metas[i].m_Type);
                 }
             }
+            else if (!prototype->m_RenderResources.Full())
+            {
+                result = dmResource::RESULT_OUT_OF_RESOURCES;
+            }
```

Comment:
This is from the old code, but I feel it's a bit odd to return this result? The reason the fail condition has happened is because some resource couldn't be loaded afaik